### PR TITLE
Added a new light layer debug mode

### PR DESCRIPTION
**Pull request by adrien-de-tocqueville**
**DONT FORGET TO ADD A CHANGELOG**

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-2019.3` label. After you backport the PR, the label changes to `backported-2019.3`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [x] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [x] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
This PR is adding a new lighting debug mode to visualize which light layers are assigned to each objects and which objects are affected by a light / are casting shadows for a light.
Light layers have to be enabled in the HDRP Asset.

New menu entries for the debug mode

Example display in a scene

The PR also reorganizes the items in the lighting debug menu in categories (Shadows, Lighting, Material Overrides, and the rest).

image

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/HDRP%252Flight-layer-debug

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by TomasKiniulis** [comment]
Looks smooth! Checked different light types with various Filter Layers
g9yStf5aes
Checked "Use Selected light" functionality
VAvRLgWI0y
Checked "Match Light's Shadow Layer" functionality
HbNoMg0Bk9

There seems to be an issue with Undo-Redo functionality not taking the debug mode into account anymore
NIuuUtS0ur
in the gif I Undo-Red "Match Light's Shadow Layers" but the view does not update anymore.

Also, after discussing with Adrian, might be good to change "Match Light's Shadow Layer" to "Switch to Light's Shadow Layers", seems easier to comprehend while "match" feels for me like it still remembers light layers and also adjusts to shadow layers.

File: com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl [hunk 1]
```diff
@@ -24,6 +24,11 @@ void GetSurfaceAndBuiltinData(FragInputs input, float3 V, inout PositionInputs p
     ZERO_INITIALIZE(BuiltinData, builtinData); // No call to InitBuiltinData as we don't have any lighting
     builtinData.opacity = alpha;
 
+#if defined(DEBUG_DISPLAY)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl]: Why is this needed? I'm not sure we want to debug light layers for unlit materials.

Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/Unlit/UnlitData.hlsl]: Seb tells me it's for shadow layers that are indeed supported for unlit. Maybe add an explicit comment about this.

File: com.unity.render-pipelines.high-definition/Runtime/Debug/LightingDebug.cs [hunk 2]
```diff
@@ -150,6 +169,20 @@ public bool IsDebugDisplayEnabled()
         public DebugLightFilterMode debugLightFilterMode = DebugLightFilterMode.None;
         /// <summary>Current Full Screen Lighting debug mode.</summary>
         public DebugLightingMode    debugLightingMode = DebugLightingMode.None;
+        /// <summary>True if Light Layers Visualization is enabled.</summary>
+        public bool                 debugLightLayers = false;
+        /// <summary>1 if Light Layers Visualization uses selected light, 2 if uses light's shadow layers.</summary>
+        public int                  debugLightLayersUseSelection = 0;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Debug/LightingDebug.cs]: Using a single boolean for two different is kind of confusing. We should have two different parameters: debugLightLayersUseSelection and debugShadowLayers or something along those lines
